test/performance/performance.go

```diff
@@ -0,0 +1,389 @@
+
+	var plc_filename, output_filename string
+	var n_per_batch, n_total, per_batch_sleep int
+	var insecure bool
+
+	pflag.StringVarP(&plc_filename, "policy", "p",
+		"/resources/templates/cfgmap-plc.yaml", "path to test policy YAML, scoped to performance directory")
+	pflag.IntVarP(&n_per_batch, "policies-per-batch", "b", 100, "number of policies to create per batch")
+	pflag.IntVarP(&n_total, "total-policies", "t", 1000, "total number of policies created")
+	pflag.IntVarP(&per_batch_sleep, "sleep", "s", 20, "time (min) to sleep after creating a batch of policies")
+	pflag.StringVar(&output_filename, "csv", "results.csv", "path to CSV to export results to")
+	pflag.BoolVar(&insecure, "insecure-skip-verify", false, "skip certificate verification on metrics requests")
+
+	pflag.Parse()
+
+	// wait for config map to be applied properly before executing command
+	secret_output, err := exec.Command("kubectl", "get", "secret", "-n", "openshift-user-workload-monitoring").CombinedOutput()
+	if err != nil || !strings.Contains(string(secret_output), "prometheus-user-workload-token") {
+		klog.V(2).Info("Setting up secret for k8s metrics")
+		_, err = exec.Command("kubectl", "apply", "-f", path.Join(performance_dir, "resources/setup/metrics-configmap.yaml")).CombinedOutput()
+		if err != nil {
+			klog.Exitf("Error applying metrics configMap: %s", err)
+		}
+		// wait for secret to be created for 2 minutes
+		found := false
+		for i := 0; i < 120; i += 5 {
+			secret_output, err = exec.Command("kubectl", "get", "secret", "-n", "openshift-user-workload-monitoring").CombinedOutput()
+			if err == nil && strings.Contains(string(secret_output), "prometheus-user-workload-token") {
+				found = true
+				klog.V(2).Info("Secret found! Waiting an additional 30 seconds for metrics setup to complete.")
+				time.Sleep(5 * time.Second)
+				break
+			}
+			klog.V(2).Info("Secret not found, waiting 5 seconds...")
+			time.Sleep(5 * time.Second)
+		}
+
+		if !found {
+			klog.Fatal("Error: Prometheus secret creation timed out")
+		}
+	}
+	secret_finder := regexp.MustCompile("prometheus-user-workload-token-[a-z0-9]{5}")
+	secret := secret_finder.FindString(string(secret_output))
+	if secret == "" {
+		klog.Exit("Error getting Prometheus metrics secret: secret not found in openshift-user-workload-monitoring ns", err)
+	}
+
+	token_output, err := exec.Command("kubectl", "get", "secret", secret, "-n", "openshift-user-workload-monitoring", "-ojson").CombinedOutput()
+	if err != nil {
+		klog.Exitf("Error getting API token from secret: %s", err)
+	}
+	var prometheus_secret map[string]interface{}
+	if err := json.Unmarshal(token_output, &prometheus_secret); err != nil {
+		klog.Exitf("Error unmarshalling secret %s to JSON", secret, err)
+	}
+	token, err := base64.StdEncoding.DecodeString(prometheus_secret["data"].(map[string]interface{})["token"].(string))
+	if err != nil {
+		klog.Exitf("Error decoding token from secret %s", secret, err)
+	}
```

Comment:
Could you please move this section of code to a helper function like `getPrometheusToken`?